presto/dbapi.py

```diff
@@ -236,10 +245,111 @@ def setinputsizes(self, sizes):
     def setoutputsize(self, size, column):
         raise presto.exceptions.NotSupportedError
 
-    # TODO support params
-    def execute(self, operation):
-        self._query = presto.client.PrestoQuery(self._request, sql=operation)
-        result = self._query.execute()
+    def _execute_prepare_statement(self, operation, statement_name):
+        """
+        Prepends the given `operation` with "PREPARE <statement_name> FROM" and
+        executes as a prepare statement.
+
+        :param operation: sql to be executed.
+        :param statement_name: name that will be assigned to the prepare
+            statement.
+
+        :raises presto.exceptions.FailedToObtainAddedPrepareHeader: Error raised
+            when unable to find the 'X-Presto-Added-Prepare' for the PREPARE
+            statement request.
+
+        :return: string representing the value of the 'X-Presto-Added-Prepare'
+            header.
+        """
+        sql = 'PREPARE {statement_name} FROM {operation}'.format(
+            statement_name=statement_name,
+            operation=operation
+        )
+
+        # Send prepare statement. Copy the _request object to avoid poluting the
+        # one that is going to be used to execute the actual operation.
+        query = presto.client.PrestoQuery(copy.deepcopy(self._request), sql=sql)
```

Comment:
Can't we use `self._create_request()` here?
Is it important to call `PREPARE` using request containing information regarding current transaction?

Reply:
The reason why I decided to do it this way is that the `Cursor` class receives a request object which can technically be different from what `_create_request` originally returns, if we assume that `Cursor` class can be instantiated directly. Also `Cursor` does no have access to the `_create_request` function since that one belongs to the `Connection` class.